util/db/cluster.go

```diff
@@ -202,7 +202,7 @@ func (db *db) getClusterSecret(server string) (*apiv1.Secret, error) {
 	}
 	srv := strings.TrimRight(server, "/")
 	for _, clusterSecret := range clusterSecrets {
-		if strings.TrimRight(string(clusterSecret.Data["server"]), "/") == srv {
+		if strings.TrimRight(string(clusterSecret.Data["server"]), "/") == srv || string(clusterSecret.Data["name"]) == server {
```

Comment:
In case if secret will not contain a name and you will pass the server as an empty string you may fetch incorrect secret



Test that repro it 

func TestDeleteClusterByNameWithUnknownName(t *testing.T) {
	kubeclientset := fake.NewSimpleClientset(&v1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "mycluster",
			Namespace: fakeNamespace,
			Labels: map[string]string{
				common.LabelKeySecretType: common.LabelValueSecretTypeCluster,
			},
		},
		Data: map[string][]byte{
			"server": []byte("http://mycluster"),
		},
	})
	settingsManager := settings.NewSettingsManager(context.Background(), kubeclientset, fakeNamespace)
	db := NewDB(fakeNamespace, settingsManager, kubeclientset)
	assert.Nil(t, db.DeleteCluster(context.Background(), ""))
}